doc/dev/lib.xml

```diff
@@ -12,6 +12,61 @@ kernel among the developers.
 
 <Chapter Label="Chap-Lib">
 <Heading>Maintaining the &GAP; kernel and library</Heading>
+
+<Section Label="Sect-debugging">
+<Heading>Debugging the GAP Kernel</Heading>
+
+The GAP kernel supports a variety of options which enabled when running <C>configure</C>,
+to help writing and debugging GAP kernel code.
+<P/>
+
+The first option, <C>--enable-debug</C>, is well supported can can be used
+whenever the GAP kernel, or kernel modules, are being developed (although
+developers should still check their code works without it).
+<P/>
+<C>--enable-debug</C> enables assertions throughout the kernel. These
+assertions are implemented using the <C>GAP_ASSERT</C> macro. These
+assertions are disabled when GAP is compiled without <C>--enable-debug</C>.
+When <C>--enable-debug</C> is enabled, <C>KernelDebug</C> will be shown in
+the Configuration line when GAP is started.
+<P/>
+
+The next two options are more experimental, and are not designed to be constantly
+enabled. They are particularly designed to track down bugs relating to memory
+corruption relating to GASMAN (GAP's memory manager). They cannot both be enabled
+at the same time.
+<P/>
+<C>--enable-memory-checking</C> makes GAP check for C pointers to the content of Bags
```

Comment:
I'd not capitalize "Bag" here, though one might want to insert an explanation of what a bag is. Then again, it makes little sense to use this option if one does not know what a GAP "bag" is. Perhaps Insert in the previous line something like this:
> The following explanations assume you are somewhat familiar with the internals of GASMAN (see e.g. <F>src/gasman.c</F> for more information).